src/Chapter2.hs

```diff
@@ -842,7 +889,11 @@ list.
 
 🕯 HINT: Use the 'cycle' function
 -}
-rotate = error "rotate: Not implemented!"
+
+rotate :: Int -> [a] -> [a]
+rotate n l
+  | n < 0 = []
+  | otherwise = take (length l) (drop n (cycle l))
```

Comment:
That's the correct solution! 🌞 

Though, there is a way to improve it a bit 💠  
It does a little bit more than work when the given number is big and the list is small. You will cycle through the list a lot. Maybe you can optimize this using `mod`, for example :slightly_smiling_face: